Sources/Sentry/SentryProfiler.mm

```diff
@@ -283,20 +285,61 @@ - (instancetype)initWithHub:(SentryHub *)hub
         return nil;
     }
 
+    _profileId = [[SentryId alloc] init];
+
     SENTRY_LOG_DEBUG(@"Initialized new SentryProfiler %@", self);
     _debugImageProvider = [SentryDependencyContainer sharedInstance].debugImageProvider;
     _hub = hub;
+    [self start];
+    [self scheduleTimeoutTimer];
+
+#    if SENTRY_HAS_UIKIT
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(backgroundAbort)
+                                                 name:UIApplicationWillResignActiveNotification
+                                               object:nil];
+#    endif // SENTRY_HAS_UIKIT
+
```

Comment:
Lines 293-301 were originally in the class method `+ [SentryProfiler startWithHub:]`.